medmodels/statistic_evaluations/comparer/data_comparer.pyi

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import polars as pl
+
+from medmodels.medrecord.medrecord import MedRecord
+from medmodels.medrecord.querying import NodeOperation
+from medmodels.medrecord.types import (
+    Group,
+    GroupInputList,
+    MedRecordAttribute,
+    MedRecordAttributeInputList,
+)
+from medmodels.statistic_evaluations.comparer.builder import ComparerBuilder
+
+class DataComparer:
+    patients_group: Group
+    time_attribute: MedRecordAttribute
+    concept_groups: GroupInputList
+    top_k_concepts: int
+    alpha: float
+    subcohorts_groups: Optional[GroupInputList]
+    subcohorts_queries: Optional[List[NodeOperation]]
+    control_cohort: Optional[Union[Group, NodeOperation]]
```

Comment:
this (i think) requires that the medrecords that are compared have the exact same group names which is not great imo

Reply:
In my opinion all the comparer functions should be staticmethods and the configuration is set for each medrecord when it is passed.

e.g. the Configuration for a medrecord would be a typed dict
```
class DataComparerConfig(TypedDict):
    patients_group: Group
    time_attribute: MedRecordAttribute
    ...
```

and the function signature for example for `calculate_distance_concepts` woudl be
```
@staticmethods
def calculate_distance_concepts(
    self, medrecords: Union[List[Tuple[MedRecord, DataComparerConfig]], Tuple[MedRecord, DataComparerConfig]]
) -> Dict[str, float]: ...
```

Can be discussed during our weekly